### PR TITLE
Refactor/97 thrusterPlatformReference refactor

### LINKED ISSUE #97: thrusterPlatformReference refactor
Refactoring the thrusterPlatformReference module in order to read in a message containing the static information on thrust vector and application point in platform frame coordinates, and outputting the same information in body frame coordinates and with respect to the origin of the B frame.

### PR DESCRIPTION
**Pull request by rcalaon**
* **Tickets addressed:** #97 
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This refactor consists in providing the thrusterPlatformReference module with an input thruster configuration msg containing the configuration information of the thruster with respect to the platform frame. A similar module is provided as an output, containing the reference thruster configuration info in body-frame coordinates. Commit 1 implements these changes in the module. Commits 2 and 3 update the documentation and unit test, respectively.

## Verification
Unit test is updated to test the correctness of input and output messages.

## Documentation
Documentation is updated to reflect the changes.

## Future work
No future work anticipated at this stage.

### CODE REVIEW

File: src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c [hunk 1]
```diff
@@ -91,10 +96,13 @@ void Update_thrusterPlatformReference(ThrusterPlatformReferenceConfig *configDat
     double r_CM_M[3];
     v3Add(r_CB_M, configData->r_BM_M, r_CM_M);               // position of C w.r.t. M in M-frame coordinates
     double r_TM_F[3];
-    v3Add(configData->r_FM_F, configData->r_TF_F, r_TM_F);   // position of T w.r.t. M in F-frame coordinates
+    v3Add(configData->r_FM_F, thrusterConfigFIn.rThrust_B, r_TM_F);   // position of T w.r.t. M in F-frame coordinates
```
Comment (patkenneally) [inline src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c:99]: Just so I know I'm following, in this message instance we are taking the `_B` frame in `rThrustB` to really denote the F frame?

Comment (rcalaon) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c:99]: That is correct, and I agree that it is not very graceful. I believe it's been done before with thrusters attached to other components, but I'm open to suggestions.

Comment (patkenneally) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c:99]: I think it's fine. We can't do much about the variable names inside the message. Just wanted to make sure.

Comment (rcalaon) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/thrusterPlatformReference.c:99]: Right. I tried to be very explicit about this in the documentation.

File: src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/_UnitTest/test_thrusterPlatformReference.py [hunk 2]
```diff
@@ -229,6 +237,21 @@ def platformRotationTestFunction(show_plots, delta_CM, K, seed, accuracy):
         testFailCount += 1
         testMessages.append("FAILED: " + platformWrap.ModelTag + "thrusterPlatformReference module failed unit test on thruster torque \n")
 
+    # compare the module results to the python computation for thruster configuration in B frame
+    r_TB_B = r_CB_B - np.matmul(FB.transpose(), r_CT_F)
+    r_TB_B_sim = thrConfigBLog.rThrust_B[0]
+    tHat_B_sim = thrConfigBLog.tHatThrust_B[0]
+    tMax_sim = thrConfigBLog.maxThrust[0]
+    if not unitTestSupport.isVectorEqual(r_TB_B_sim, r_TB_B, accuracy):
```
Comment (patkenneally) [inline src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/_UnitTest/test_thrusterPlatformReference.py]: Hit me with some `np.testing` assert sweetness and then we can merge.

Comment (rcalaon) [reply src/fswAlgorithms/effectorInterfaces/thrusterPlatformReference/_UnitTest/test_thrusterPlatformReference.py]: On it 🔥

**Review by patkenneally — APPROVED** [review]

**Pull request merged**